### DISCUSSION TITLE
Mongoose Type 'Promise<Document>' is not assignable to type 'Promise<Apartment>'.

### DISCUSSION THREAD

**Discussion by micgre0** [dotansimha/graphql-code-generator#5105] [Q&A]
Hi, I use Mongoose, and I want to return from the resolver created Document type but generated resolver types making it impossible for me. Does anyone know how to get around the issue or make it works? Plugins? Thank you in advance.
--------------------------------------------------------------------------------------------------------------------------------
    Type 'Promise<Document>' is not assignable to type 'Apartment | Promise<Apartment> | Promise<ResolverTypeWrapper<Apartment>>'.
      Type 'Promise<Document>' is not assignable to type 'Promise<Apartment>'.
        Type 'Document' is missing the following properties from type 'Apartment': location, category, title, description, and 4 more.
--------------------------------------------------------------------------------------------------------------------------------
type Apartment {
    _id: ID!
    location: String!
    category: String!
    title: String!
    description: String!
    rooms: Int!
    area: Float!
    price: Float
    priceCurrency: Float
    floor: Int!
    numberOfFloors: Int!
  }
"@graphql-codegen/cli": "^1.19.2",
"@graphql-codegen/typescript": "^1.17.11",
"@graphql-codegen/typescript-resolvers": "^1.17.11",
"mongoose": "^5.10.9"

**Comment by ardatan** [comment]
You can use `mappers` field to map `Apartment` to `Document` inside resolvers;
https://graphql-code-generator.com/docs/plugins/typescript-resolvers
```yml
config:
   mappers:
        Apartment: Document
```

**Reply by micgre0** [reply]
Apartment: import("mongoose").Document - This works but when I try 
```
apartments: () => {
      return Apartment.find({});
    }, 
```
------------------------------------------------
```
Type '() => DocumentQuery<Document[], Document, {}>' is not assignable to type 'ResolverFn<Maybe<ResolverTypeWrapper<Document>[]>, {}, any, {}>'.
    Type 'DocumentQuery<Document[], Document, {}>' is not assignable to type 'ResolverTypeWrapper<Document>[] | Promise<Maybe<ResolverTypeWrapper<Document>[]>> | null | undefined'.
      Type 'DocumentQuery<Document[], Document, {}>' is missing the following properties from type 'ResolverTypeWrapper<Document>[]': length, pop, push, concat, and 22 more.
```
------------------------------------------------

**Comment by micgre0** [comment]
I got it works but I don't know is it the correct solution?
`import { Apartment as ApartmentType } from "./resolvers-types";`
Then
```
apartments: () => {
      return Apartment.find({}) as unknown as ApartmentType[];
    },
```
-------------------------------------------------------------------
```
createApartmentOffer: async (_, args) => {
      ...

      const apartment = new Apartment({
        location,
        category,
        title,
        description,
        rooms,
        area,
        price,
        priceCurrency,
        floor,
        numberOfFloors,
      });

      try {
        await apartment.save();

        return apartment as unknown as ApartmentType;
      } catch (error) {
        ...
      }
    },
  },
```
